test/mocks/ticker.go

```diff
@@ -0,0 +1,78 @@
+package mocks
+
+import (
+	"sync"
+	"time"
+)
+
+func NewTicker() *Ticker {
+	now := time.Now()
+
+	ticker := &Ticker{
+		sigTime:  make(chan time.Time),
+		sigClose: make(chan struct{}, 1),
+		sigAdd:   make(chan time.Duration),
+		sigReset: make(chan time.Duration),
+		d:        1000 * time.Hour,
+		ch:       make(chan time.Time, 1),
+		time:     now,
+		lastTick: now,
+	}
+
+	return ticker
+}
+
+type Ticker struct {
+	lock     sync.RWMutex
+	sigTime  chan time.Time
+	sigClose chan struct{}
+	sigReset chan time.Duration
+	sigAdd   chan time.Duration
+	d        time.Duration
+	ch       chan time.Time
+	time     time.Time
+	lastTick time.Time
+}
+
+func (t *Ticker) Stop() {
+	t.lock.Lock()
+	defer t.lock.Unlock()
+
+	close(t.sigTime)
+	close(t.sigAdd)
+	close(t.sigReset)
+	close(t.sigClose)
+}
+
+func (t *Ticker) Channel() <-chan time.Time {
+	return t.ch
+}
+
+func (t *Ticker) Now() time.Time {
+	t.lock.Lock()
+	defer t.lock.Unlock()
```

Comment:
```suggestion
	t.lock.RLock()
	defer t.lock.RUnlock()
```
since rw mutex has been used and here there is only reading